---
title: "Brief increases in corticosterone result in immediate and lasting changes to DNA methylation in a wild bird"
author: 
  - Conor C. Taff
  - Sabrina M. McNew
  - Leonardo Campagna
  - Maren N. Vitousek
header-includes:
  - \usepackage[font={footnotesize}, labelfont={bf}]{caption}
  #- \usepackage{setspace}\doublespacing
  #- \usepackage{lineno}\linenumbers
fontsize: 12pt
output: 
  bookdown::pdf_document2:
      number_sections: FALSE
      toc: FALSE
      extra_dependencies: ["flafter"]
  bookdown::word_document2:
      number_sections: FALSE
      toc: FALSE
      extra_dependcies: ["flafter"]
  bookdown::html_document2:
      number_sections: FALSE
      toc: TRUE
bibliography: references.bib
csl: molecular-ecology.csl
---

```{r setup, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
```

## ABSTRACT



*Keywords: reduced representation bisulfite sequencing, stress, endocrinology, carryover effects*

## INTRODUCTION

Wild animals regularly encounter challenging conditions that require rapid behavioral and physiological responses. In vertebrates, the glucocorticoid mediated stress response plays an essential role in allowing animals to successfully avoid or tolerate stressors [@sapolsky2000; @wingfield1998]. While an appropriate response is beneficial [@wingfield1998], an inappropriate or prolonged elevation of glucocorticoids can result in a variety of well described costs [@korte2005]. Given the way that these benefits and costs can change over time or between contexts [cite], flexible regulation of the acute stress response may itself be an important determinant of coping ability [@taff2016; @hau2016; @wada2014]. Measuring these within-individual reaction norms in endocrine traits is challenging, but has the potential to clarify why some individuals succeed when others fail during challenging conditions. However, a full understanding of how flexibility contributes to coping ability also requires identifying the mechanism(s) that contribute to flexible adjustment of the acute stress response. 

Changes in DNA methylation in response to prior challenges represent one mechanism that could play a role in the calibration of stress response systems. Epigenetic modification by DNA methylation can alter phenotypes by making genes or promoters more or less accessible for transcription [cite]. It is well known that early life experiences can have profound programming effects on DNA methylation patterns that often persist throughout the lifetime [cite]. For example, classic work in lab rodents demonstrates that early life experiences regulate methylation of the gene producing the glucocorticoid receptor, which results in lifelong changes to glucocorticoid secretion in response to challenges [cite]. A growing number of studies also demonstrate early life programming of DNA methylation patterns in wild populations resulting from dominance hierarchies [@laubach2019], brood size [@jimeno2019].

While they are less well documented, experiences during adulthood can also result in changes in DNA methylation and these adjustments can occur rapidly [cite]. For example, brief periods of experimental competition and aggression in tree swallows (*Tachycineta bicolor*) resulted in altered DNA methylation of brain regions associated with hormone signaling, suggesting a priming effect in preparation for future aggression [@bentz2021]. Conceptual models of the stress response have long recognized that the sequence, frequency, duration, and intensity of stressors should change the optimal behavioral and physiological response [romero, wingfield, taff, others]. Yet it is often unclear how the experience of challenges during adulthood would be biologically encoded to alter responses to future challenges. Altered DNA methylation is a promising mechanism because it i) can change rapidly even during adulthood, ii) changes can persist over moderate to long time scales, iii) has been shown to change with challenging experiences, and iv) may play a direct role in physiological responses to future challenges. It has not yet been demonstrated, however, whether brief increases in corticosterone as an adult causally influence methylation patterns.

We studied covariation between DNA methylation and corticosterone (the primary avian glucocorticoid) using both observational and experimental approaches. In this population we previously found that genome wide methylation predicts resilience to experimental challenges [@taff2019] and that brief experimental increases in corticosterone have lingering effects on behavior and performance [@vitousek2018; @taff2018]. However, corticosterone and methylation have not been studied together and the methylation methods used previously did not allow for base-pair level resolution of changes in methylation status. Here, we used reduced representation bisulfite sequencing (RRBS) and a newly improved reference genome sequenced for this study to examine genome wide patterns of DNA methylation at high resolution. We first assessed covariation between methylation and natural variation in corticosterone regulation during an acute stress response. Next, we experimentally simulated a series of acute corticosterone responses using a non-invasive dosing procedure [@vitousek2018] and compared DNA methylation to controls to determine whether brief increases in corticosterone resulted in altered DNA methylation at either short (1-2 weeks) or long (1 year) timescales. 

Given previous work demonstrating a correlation between coping ability and both genome-wide methylation [@taff2019] and natural variation in rapid corticosterone regulation [@vitousek2018f], we predicted that natural variation in corticosterone (baseline and stress-induced increase) would be associated with DNA methylation. However, a correlation here could arise through early life programming, prior activation of the acute corticosterone response, or any conditions that impact the regulation of both methylation and corticosterone (e.g., body condition). In contrast, for the experimental manipulation we predicted that differences in DNA methylation between control and treatment groups would only be present if brief increases in corticosterone have a causal effect on altering methylation patterns. We assessed the time course and persistence of any such changes using comparisons 1-2 weeks after treatments and 1 year after treatments. If methylation changes play a role in altering future corticosterone secretion then we expected to find more differences near genes and promoters associated with endocrine regulation. 

## METHODS

We studied tree swallows breeding at field sites in and around Ithaca, New York, U.S.A. from April to July of 2015 and 2016. This population of tree swallows has been continuously studied since 1986 and we followed well established monitoring protocols [for details see @winkler2020]. In 2015, adult females were captured on day **X** after the beginning of incubation and again on day **XX** of incubation.  In 2016, any returning females were captured on day **XX** of incubation. At each capture we collected three blood samples (< 70$\mu$l each) to measure baseline (< 3 minutes), stress induced (30 minutes), and post negative feedback (30 minutes after dexamethasone injection) corticosterone [@vitousek2018]. We also collected a set of standardized morphological measurements and monitored reproductive success [@vitousek2018]. All birds received a unique USGS aluminum band and passive integrated transponder (PIT) tag if they were not previously banded.

Between the first and second capture in year one, females were randomly assigned to either a control or experimental treatment group. In the experimental group, we simulated a brief spike in corticosterone on 5 days between the two captures. To accomplish this, we applied a 60 $\mu$l dose of corticosterone gel (xx DMSO, xx corticosterone: NEED TO CHECK TREATMENTS THERE ARE ACTUALLY SEVERAL DIFFERENT CORT DOSES) to a fake egg anchored in the nest cup at a randomly chosen time during the day when females were absent from the nest. Upon returning, females incubated the clutch and absorbed a dose of corticosterone across the brood patch. 

We validated that this dosing method results in a brief (< 180 minutes) increase in corticosterone within the range of natural acute corticosterone responses [@vitousek2018]. Control nests received either no manipulation or a sham control in which they were dosed as described above but with DMSO gel only with no corticosterone added. We previously found no difference in physiology, behavior, reproductive success, or survival between control and sham control birds receiving this treatment [@vitousek2018; @taff2018] and we combine both control groups in the analyses described here. 

*Tree swallow reference genome assembly*

For this study, we improved upon a previously published reference genome sequenced from a female from this study population [@taff2019]. We extracted fresh DNA from the same individual and submitted to Duke(?) for xxx sequencing. We combined the new sequence data with the xxx from previous work. List of tools and process for assembly. List of some characteristics of new genome, where to access, etc.

*Sample processing*

Blood samples collected in the field were immediately stored on ice in a cooler and processed in the lab within 3 hours of capture. Red blood cells were separated from plasma by centrifugation and added to 1 mL of ice cold cryopreservation buffer [90% newborn calf serum, 10% DMSO, @haussmann2008]. Samples were then frozen at a constant cooling rate in a Mr. Frosty container with isopropyl alcohol and stored at -80$^\circ$ C until further processing. Cryopreserved blood samples were thawed and DNA was extracted using the DNeasy Blood & Tissue spin column extraction kits according to the manufacturer's protocol. 

*Reduced representation bisulfite sequencing*

We prepared our samples for reduced representation bisulfite sequencing (RRBS) using the Diagenode Premium RRBS Kit and closely following the manufacturer's protocol [@veillard2016]. Briefly, samples were diluted to **XX** and **XX** of sample was used for library preparation. The process included enzymatic digestion with Mspl and size selection to increase coverage of CpG-rich regions, such as CpG islands and enhancers. Individual samples received a unique barcode and were pooled in groups of 8 before bisulfite conversion. We also included a methylated and unmethylated spike in control with each sample to confirm the efficiency of bisulfite conversion.

From the available samples, we selected 120 samples to process. These samples were chosen to maximize the power of our planned comparisons (i.e., preferentially birds with all 3 samples, then birds with just two samples). Prior to RRBS processing, these 120 samples were assigned randomly sorted for processing to account for any batch effects. Libraries were prepared with the Diagenode kit in two batches (one set of 24 and one of 96). Prepared libraries were checked for the expected size distribution by digital PCR prior to sequencing. Sequencing was performed at the Cornell BRC on a NextSeq 1x75 run with 20% PhiX and 85% of the normal cluster density. In total, we ran our samples on five sequencing lanes with 24 samples per lane.

Raw sequence data were first processed with `Trim Galore!` using the default RRBS settings. Visual inspection of FastQC files confirmed high quality reads for all samples. Next, we used `Bismark` to align each sequence to the prepared genome and extract the methylation status for each CpG, CpH, or CHH site [@krueger2011]. As expected, global methylation at CpH and CHH sites was extremely low (1% and 0.6%, respectively) and we only considered methylation at CpG sites in our subsequent analyses [e.g., xx]. We also used `Bismark` to determine the methylation conversion efficiency for each sample based on methylated and unmethylated spike in controls and following the instructions in the Diagenode RRBS kit [@veillard2016; @krueger2011]. 

*Data analysis*

Output data from the sequence processing described above was analyzed in R version 4.1.1 [@rcite]. We initially processed the aligned sequence data using `MethylKit`, which includes a variety of functions for working with bisulfite sequencing data [@akalin2012]. Using `MethylKit`, we extracted the number of total aligned reads and number of methylated or unmethylated reads for each CpG site that met the following criteria. First, we required a minimum coverage of 10 reads per sample and 10 samples per group to retain a CpG. We used this full set of CpGs for summary statistics, but for modeling the effects of treatments we further filtered the dataset to remove any CpGs that were mostly invariant (more than half of samples had methylation percentage of 0 or 100%) as well as CpGs that had extremely low variation (SD less than 5% across all samples). 

Filtering conducted separately for pre/post in year vs. y1/y2. We used the pre samples only to look for natural correlations with cort

The built in differential methylation techniques in `MethylKit` are designed for two group comparisons with limited flexibility in modeling options and no ability to include random effects. Because we had repeated measures before and after treatments for both groups, we could not specify the necessary models within `MethylKit` itself. Therefore, we exported and combined the filtered CpG records for all groups so that we could fit generalized linear mixed models (GLMMs) [as in VAN OERS].

We fit a similar set of models for natural corticosterone variation, within-year treatments, and between-year treatments; each of these datasets were constructed separately since they included different subsets of both individual birds and of CpGs that met the criteria described above. For natural variation in corticosterone, we fit a generalized linear model (GLM) with the number of methylated and unmethylated reads as the binomial response variable. Predictors included baseline, stress-induced, and post negative feedback corticosterone. Because these models used pre-treatment samples, only a single observation of each individual was included and there was no need to include random effects.  

For within year comparisons, we fit a single GLMM for each included CpG with the number of methylated and unmethylated reads as the binomial response variable. Predictors included each of the four treatment categories (control pre and post; corticosterone-dosed pre and post) along with a random effect for individual identity. Between year comparisons were similar except that only two groups were included as fixed effects (one year post corticosterone vs control). In both cases, significance of comparisons between each group of interest was assessed by calculating least squared means from fit models using the `emmeans` package in R [@emmeans]. We accounted for multiple comparisons in each of these models by adjusting all p-values using the q-value approach implemented by the `qvalue` package in R with the false discovery rate set at 0.05 [@qvalue]. We subsequently relaxed this false discovery rate to include marginally different CpGs in our gene ontology analyses.

The complete set of R code and sample data is available on GitHub and will be publicly archived upon acceptance (www.github.com/cct663/tres_rrbs). Raw sequence data from reduced representation bisulfite sequencing is available at (where to host this? sra?). The assembled genome used for sequence alignment is available at (??). 

## RESULTS

*RRBS results*

Our process resulted in 9.8 $\pm$ 4.3 million (SD) total reads per sample (Figure S1). Across all samples, we were able to align 51.1% of the total reads produced, which is comparable to several recent studies in wild birds [cite]. Spiked controls in each sample indicated that our bisulfite conversion worked efficiently within the recommended kit parameters (conversion of methylated control sites = 1.9% $\pm$ 1.4; unmethylated control sites = 99.5% $\pm$ 0.6). 

*Within-year effect of corticosterone*

*Between year effect of corticosterone*

## DISCUSSION

## ACKNOWLEDGMENTS

## AUTHOR CONTRIBUTIONS

## ETHICAL NOTE

## FUNDING

## REFERENCES