---
title: "Corticosterone exposure is associated with long-term changes in DNA methylation, physiology, and breeding decisions in a wild bird."
author: 
  - Conor C. Taff
  - Sabrina M. McNew
  - Leonardo Campagna
  - Maren N. Vitousek
header-includes:
  - \usepackage[font={footnotesize}, labelfont={bf}]{caption}
  - \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
  - \usepackage{setspace}\doublespacing
  - \usepackage{lineno}\linenumbers
fontsize: 12pt
output: 
  bookdown::pdf_document2:
      number_sections: FALSE
      toc: FALSE
      extra_dependencies: ["float"]
  bookdown::word_document2:
      number_sections: FALSE
      toc: FALSE
      extra_dependcies: ["flafter"]
  bookdown::html_document2:
      number_sections: FALSE
      toc: TRUE
bibliography: references.bib
csl: molecular-ecology.csl
always_allow_html: true
---

```{r setup, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
```

\raggedright

*CCT, MNV, & LC: Department of Ecology & Evolutionary Biology and Lab of Ornithology, Cornell University* 

*SMM: Department of Ecology & Evolutionary Biology, University of Arizona* 

*Correspondence: Conor Taff, cct63@cornell.edu*

\pagebreak

## ABSTRACT

When facing challenges, vertebrates activate an evolutionarily conserved hormonal stress response that can dramatically alter behavior and physiology. Although this response can be costly, conceptual models suggest that it can also recalibrate the stress response system, priming more effective responses to future challenges. Little is known about whether this process occurs in wild animals, particularly in adulthood, and if so, how information about prior experience with stressors is encoded. One potential mechanism is hormonally-mediated changes in DNA methylation. We simulated the spikes in corticosterone that accompany an acute stress response using non-invasive dosing in female tree swallows (*Tachycineta bicolor*) and monitored the phenotypic effects one year later, and DNA methylation both shortly after treatment and a full year later. The year after treatment, experimental females had stronger negative feedback and initiated breeding earlier – traits that are associated with stress resilience and reproductive performance in our population – and higher baseline corticosterone. We also found that natural variation in corticosterone predicted patterns of DNA methylation.  Finally, corticosterone treatment influenced methylation on short (1-2 weeks) and long (1 year) time scales; however, these changes did not have clear links to functional regulation of the stress response. Taken together, our results are consistent with corticosterone-induced priming of future stress resilience and support DNA methylation as a potential mechanism, but more work will be needed to demonstrate functional consequences. Uncovering the mechanisms linking experience with the response to future challenges has implications for understanding the drivers of stress resilience.

*Keywords: reduced representation bisulfite sequencing, stress, endocrinology, carryover effects, tree swallow*

```{r dont-run, echo = FALSE, message = FALSE, warning = FALSE}
# not needed for this submission

### SIGNIFICANCE

#A stress response to an environmental challenge can preserve an individual's fitness and may even prime them to survive similar challenges in the future. What mechanisms underlie priming is unclear, but epigenetic alterations to stress-related genes are one possibility. We experimentally tested whether increasing corticosterone in free-living swallows had lasting phenotypic or epigenetic effects. A year after treatment, females who received corticosterone had altered stress physiology and bred earlier than control birds, traits that are associated with higher fitness. Treatment also altered DNA methylation and methylation of the MC2R gene was correlated with stress physiology. This study adds to a growing body of literature suggesting that epigenetic changes are key to animals' response to a changing environment.
```

## INTRODUCTION

Wild organisms regularly encounter challenging conditions that require rapid behavioral and physiological responses. In vertebrates, the glucocorticoid mediated stress response plays an essential role in allowing animals to successfully avoid or tolerate stressors [@sapolsky2000; @wingfield1998]. While an appropriate response is beneficial [@wingfield1998], an inappropriate or prolonged elevation of glucocorticoids can result in a variety of negative consequences for health and fitness [@korte2005]. Accordingly, the dominant paradigm in behavioral ecology and endocrinology is that the immediate benefits of the stress response are balanced by long-term costs.

But are the long-term effects of a physiological stress response always costly? Some conceptual models propose that activating the stress response system--even in adulthood--primes more effective responses to future challenges [@hilker2016; @del2018]. These models predict that initiating a response calibrates the stress response system, increasing organismal ability to return to normal functioning after a disturbance (resilience) or the ability to maintain normal performance despite a disturbance (robustness) [@crespi2021]. Because physiological priming results from activating the stress response system rather than from learning, it could occur even in the absence of exposure to an identifiable external threat. Physiological studies have provided some evidence that stressor priming can occur, including outside of critical developmental periods [@marasco2015; @andrade2016]. However, we know little about the degree to which stressor priming operates in natural populations, affecting later life behavior, physiology, and fitness. Similarly, the mechanism(s) that link activation of the stress response to physiological regulation of subsequent responses are not well understood.

One mechanism that could play a role in the calibration of stress response systems is changes in DNA methylation. Epigenetic modification by DNA methylation can alter phenotypes by making genes or promoters more or less accessible for transcription [@lea2018; @sepers2019; @anastasiadi2018]. While the majority of methylation differences may be genetically determined [@sepers2023], early life experiences can also have profound programming effects on DNA methylation patterns that often persist throughout the individuals' lifetime [@mcgowan2009]. For example, classic work in lab rodents demonstrates that early life experiences regulate methylation of the gene producing the glucocorticoid receptor, which results in lifelong changes to glucocorticoid secretion in response to challenges [@liu1997; @weaver2004]. A growing number of studies in wild animals also report patterns consistent with early life programming of DNA methylation in wild populations associated with dominance hierarchies [@laubach2019], brood size [@sheldon2018], temperature and weather [@metzger2017; @rubenstein2016], or urbanization [@vonholdt2022].

While they are less well documented, experiences during adulthood can also result in changes in DNA methylation and these adjustments can occur rapidly [@metzger2017; @bentz2021]. For example, brief periods of experimental competition and aggression in tree swallows (*Tachycineta bicolor*) resulted in altered DNA methylation of brain regions associated with hormone signaling, suggesting a priming effect in preparation for future aggression [@bentz2021]. Conceptual models of the stress response have long recognized that the sequence, frequency, duration, and intensity of stressors should change the optimal behavioral and physiological response [@romero2009; @zimmer2022]. Yet it is often unclear how the experience of challenges during adulthood would be biologically encoded to alter responses to future challenges. Altered DNA methylation is a promising mechanism because i) it can change rapidly even during adulthood [@bentz2021; @hu2018], ii) it might persist over moderate to long time scales [@zanas2015; @lea2016], iii) it has been shown to change with challenging experiences [@hu2018; @metzger2017], and iv) it can directly alter an individual's phenotype [@sagonas2020; @angers2010].

We experimentally simulated a series of acute corticosterone spikes using a non-invasive dosing procedure [@vitousek2018] and monitored both long-term phenotypic effects and changes in DNA methylation. In this population we previously found that brief increases in corticosterone have effects on behavior and performance that persist throughout at least the length of the breeding season [@vitousek2018; @taff2018] and that genome-wide methylation predicts resilience to experimental challenges [@taff2019]. Here, we extended those results to ask whether brief increases in corticosterone altered regulation of the stress response and breeding decisions a full year later. We coupled this approach with reduced representation bisulfite sequencing (RRBS) and a newly improved reference genome assembled for this study to examine genome-wide patterns of DNA methylation at high resolution. We used RRBS to first assess covariation between methylation and natural variation in corticosterone regulation during an acute stress response. Next, we compared DNA methylation in corticosterone treated females to controls to determine whether brief increases in corticosterone resulted in altered DNA methylation at either short (1-2 weeks) or long (1 year) timescales. 

If activation of the stress response machinery primes future coping ability [@hilker2016; @del2018], experimentally treated females should exhibit long-term (across-year) differences in key phenotypic traits. Specifically, we predicted that in the year after treatment females would have a robust stress-induced increase in corticosterone and strong negative feedback, measured as the ability to quickly down regulate corticosterone after handling stress. These traits that have been previously shown to predict stress resilience in this population [@zimmer2019]. We also predicted that these females would initiate breeding earlier in the subsequent year, which is associated with higher reproductive performance in tree swallows [@winkler2020]. Finally, given previous work demonstrating a correlation between coping ability and both genome-wide methylation [@taff2019] and natural variation in rapid corticosterone regulation [@vitousek2018f], we predicted that natural variation in corticosterone (baseline, stress-induced increase, and efficacy of negative feedback) would be associated with DNA methylation. However, a correlation here could arise through early life programming, prior activation of the acute corticosterone response, or any conditions that impact the regulation of both methylation and corticosterone (e.g., body condition). In contrast, for the experimental manipulation we predicted that differences in DNA methylation between control and treatment groups would only be present if brief increases in corticosterone have an effect on altering methylation patterns. We assessed the time course and persistence of any such changes using comparisons 1-2 weeks after treatments and 1 year after treatments. If methylation changes play a role in altering future corticosterone secretion then we expected to find more differences near genes and promoters associated with endocrine regulation. 

```{r exp-fig, warning = FALSE, message = FALSE, echo = FALSE, out.width = "100%", fig.align = "left", fig.cap = "Schematic illustration of the experimental treatment and samples collected for RRBS. Models comparing natural corticosterone to DNA methylation used pre-treatment samples (A). After treatments were applied (B), models testing for within-year effects of treatment used post-treatment samples (C), while controlling for initial methylation (A). Models testing for between-year effects used post-treatment samples (D), while controlling for initial methylation (A). Models testing for between-year phenotypic effects on corticosterone adn breeding timing used a larger set of females (D). See text and table S1 for description of birds included in each group. Note that sample sizes indicate number of samples used in sequencing and analysis, but are not indicative of different between year return rates in each treatment because there were more corticosterone treatments applied."}

knitr::include_graphics("experiment_fig.png")

```

## MATERIALS AND METHODS

We studied tree swallows breeding at field sites in and around Ithaca, New York, U.S.A. from April to July 2014 to 2017. This population of tree swallows has been continuously studied since 1986 and we followed well-established monitoring protocols [for details see @winkler2020]. Adult females were captured on day 6 to 7 after the beginning of incubation and again on day 3 to 7 after eggs had hatched. In the year after treatment, any returning females were captured on day 6 to 7 of incubation. At each capture we collected blood samples (< 70$\mu$l each) to measure baseline (< 3 minutes) and stress-induced (30 minutes) corticosterone [@vitousek2018]. Immediately after the second blood sample was taken, females were injected with 4.5 $\mu$l/g of dexamethasone in the pectoralis muscle [Mylan 4mg ml^-1^ dexamethasone sodium phosphate; previously validated in @zimmer2019]. This injection stimulates strong negative feedback, which is the process resulting in downregulation of circulating corticosterone after an acute increase. A final blood sample was collected 30 minutes after injection to measure the efficacy of negative feedback. We also collected a set of standardized morphological measurements and monitored reproductive success [@vitousek2018]. All birds received a unique USGS aluminum band and passive integrated transponder (PIT) tag if they were not previously banded.

Between the first and second capture in year one, females were randomly assigned to either a control or experimental treatment group (experiment schematic and sample sizes at each stage are shown in Figure \@ref(fig:exp-fig)). We later confirmed that females in these groups did not differ significantly in any pre-treatment corticosterone measures. In the experimental group, we simulated a brief spike in corticosterone once per day on five consecutive days between the two captures. To accomplish this, we applied two 60 $\mu$l doses of corticosterone dissolved in DMSO gel one hour apart to a fake egg anchored in the nest cup at a randomly chosen time during the day when females were absent from the nest. Upon returning, females incubated the clutch and absorbed the corticosterone across the skin on their brood patch. For the purposes of this study, we considered females as part of the corticosterone treatment group if they received any of the three dose concentrations described in Vitousek et al. 2018 (high = 4 mg ml ^-1^ corticosterone once per day followed by sham once per day; low = 2 mg ml ^-1^ once per day followed by sham once per day; long = 2 mg ml ^-1^ twice per day one hour apart).

We previously validated that this dosing method results in a brief (< 180 minutes) increase in corticosterone within the range of natural acute corticosterone responses [@vitousek2018]. The elevation and time course of clearance from a single dose using this method is within the normal parameters for a natural acute corticosterone response triggered by a standard handling protocol. Control nests received either no manipulation or a sham control in which they were dosed as described above but with DMSO gel only with no corticosterone added. We previously found no difference at any time point in physiology, behavior, reproductive success, or survival between control and sham control birds receiving this treatment [@vitousek2018; @taff2018] and we combined both control groups in the analyses described here. 

For methylation analyses, we focused on the set of females that were manipulated in 2015 and--if they returned--recaptured in 2016. In analyses focused on between year effects of treatments on later corticosterone regulation and breeding decisions, we also included a smaller number of females observed from 2014 to 2015 and from 2016 to 2017. These additional samples included slight variants on corticosterone dosing that we considered part of the corticosterone treatment group [six doses or three doses of 4 mg ml^-1^ corticosterone once per day during incubation, @taff2018]. In the year after exposure, we only considered potential carryover effects of treatment on corticosterone regulation when females were captured on day 6 of incubation and on the timing of clutch initiation because most females were subsequently entered into unrelated experiments that could have influenced later season measures. The slight variations on corticosterone exposure all result in categorically different regulation of corticosterone when compared to control treatments Figure \@ref(fig:trt-plot)) and given our sample size, we only considered comparisons between control and combined corticosterone exposure treatments. The details sample sizes for each set of analyses are included in Table \@ref(tab:n_tab)).

*Tree swallow reference genome assembly*

For this study, we improved upon a previously published reference genome sequenced from a female belonging to this study population [@taff2019] by first extracting high molecular weight DNA from this same individual. We performed a phenol-chloroform extraction followed by an ethanol precipitation and finally a bead cleanup. The Duke Center for Genomic and Computational Biology core facility used the DNA to produce a large insert library (15 to 20 kb), which was subsequently sequenced on 3 cells of a Pacific Biosciences RSII instrument. This produced a total of 9.6 Gbp of data with an average read length of 12,053 bp and an N50 subread length of 15,643 bp. We used bamtools version 2.5.1 [@barnett2011] to merge the reads from the difference cells and retain only those that were longer than 4,500 bp (47.6% of the original raw reads). We improved our first assembly with the PBJelly2 module of PBSuite version 15.8.24 [@english2012], which uses long reads to fill or reduce gaps. This pipeline produced an assembly which was moderately improved from the previous version [@taff2019]. The total length of the assembly was 1.22 Gb (previously 1.14 Gb) and was contained in 49,278 scaffolds (previously 92,148), with an N50 of 82.9 kb (originally 34 kb) and 1.9% Ns (vs. 5.8%). We assessed the completeness of our assembly by running BUSCO version 5.2.2 [@simao2015], using the passeriformes dataset of 10,844 conserved genes. We found 80.5% of these genes in a single and complete copy, 3.9% were fragmented, 1.8% were duplicated, and 13.8% were missing. Finally, we annotated the genome following the pipeline described in Taff et al. [-@taff2019]. The assembly generated for this project is deposited on GenBank (BioProject ID PRJNA553513).

*Sample processing*

Blood samples collected in the field were immediately stored on ice in a cooler and processed in the lab within 3 hours of capture. Red blood cells were separated from plasma by centrifugation and added to 1 mL of ice cold cryopreservation buffer [90% newborn calf serum, 10% DMSO, @haussmann2008]. Samples were then frozen at a constant cooling rate with isopropyl alcohol and stored at -80$^\circ$ C until further processing. Cryopreserved blood samples were thawed and DNA was extracted using the DNeasy Blood & Tissue spin column extraction kits according to the manufacturer's protocol (Qiagen Sciences Incorporated). Plasma was used to measure corticosterone after a triple ethyl acetate extraction with commercially available ELISA kits following the manufacturer's protocol (DetectX Corticosterone K014-H5, Arbor Assays, Ann Arbor, MI). The protocol has been previously validated for tree swallows from this population in our lab [@vitousek2018].

*Corticosterone and breeding timing data analysis*

We used general linear models to compare corticosterone and the timing of breeding between control and experimental females one year after corticosterone manipulations. We fit four models in total with either the date of clutch initiation or circulating corticosterone levels (baseline, stress-induced, or post-dexamethasone injection) as the response variable. Predictors included treatment and year as a categorical fixed effect. The model for stress-induced corticosterone also included baseline corticosterone as a predictor because baseline levels can influence the amount of stress induced increase. The model for post-dexamethasone corticosterone also included stress-induced corticosterone as a predictor because the stress-induced level directly influences post-dexamethasone levels. 

*Reduced representation bisulfite sequencing*

We prepared our samples for reduced representation bisulfite sequencing (RRBS) in house using the Diagenode Premium RRBS Kit and closely following the manufacturer's protocol [@veillard2016]. Briefly, samples were diluted to 3.85 ng/$\mu$l and 26 $\mu$l of diluted sample was used for library preparation. The process included enzymatic digestion with Mspl and size selection to increase coverage of CpG-rich regions, such as CpG islands and enhancers. Individual samples received one of twenty-four unique barcodes and were pooled in randomized groups of 8 before bisulfite conversion. We also included a methylated and unmethylated spike in control with each sample to confirm the efficiency of bisulfite conversion.

From the available samples, we selected 120 samples to process from 61 unique birds (three samples per bird n = 14, two samples n = 31, one sample n = 16, Figure \@ref(fig:exp-fig)). We prioritized selection of individuals that had data from both years and that had complete data for the planned analyses. For birds with the same available samples, we selected which individuals to include randomly. Prior to RRBS processing, these 120 samples were randomly sorted to account for any batch effects. Libraries were prepared with the Diagenode kit in two batches (one set of 24 and one of 96). Prepared libraries were checked for the expected size distribution by digital PCR prior to sequencing. Sequencing was performed at the Cornell BRC using NextSeq sing-end 1x75 with 20% PhiX and 85% of the normal cluster density. In total, we ran our samples on five sequencing lanes with 24 samples per lane.

Raw sequence data were first processed with `Trim Galore!` version 0.6.6 using the default RRBS settings. Visual inspection of FastQC files confirmed high quality reads for all samples. Next, we used `Bismark` version 0.23.0 to align each sequence in end-to-end mode to the prepared genome and extract the methylation status for each CpG, CpH, or CHH site [@krueger2011]. As expected, global methylation at CpH and CHH sites was extremely low (1.0% and 0.6%, respectively, Figure \@ref(fig:read-summary)) and we only considered methylation at CpG sites in our subsequent analyses. We also used `Bismark` to determine the methylation conversion efficiency for each sample based on methylated and unmethylated spike in controls and following the instructions in the Diagenode RRBS kit [@veillard2016; @krueger2011]. Finally, we used `Bismark` to visually inspect m-bias plots and determined that no further trimming was necessary.

*General methylation patterns*

Our process resulted in 9.8 $\pm$ 4.3 million (SD) total reads per sample (Figure \@ref(fig:read-summary)). Across all samples, we were able to align 51.1% of the total reads produced, which is comparable to several recent studies in wild birds [@makinen2019; @watson2021]. Spiked controls in each sample indicated that our bisulfite conversion worked efficiently and within the recommended kit parameters (conversion of methylated control sites = 1.9% $\pm$ 1.4; conversion of unmethylated control sites = 99.5% $\pm$ 0.6). 

```{r no-include, echo = FALSE, message = FALSE, warning = FALSE}

# note on alignment percent for bird RRBS
  # Markinen 2019, great tit reference genome, 52 +/- 2% mapping, suggest mapping lower because of short reads or incomplete reference genome
  # Derks et al, great tit reference genome, 52% (brain) and 64% (blood) mapping
  # watson evolutionary applications, great tit 54%
  # Lundregan: biorxiv, house sparrow, 81% alignment (others to short or map to multiple regions)
  # vast majority i looked at don't list their alignment percentage

```

Among 45 pre-treatment samples, we had sufficient coverage to estimate methylation at 148,167 CpGs (minimum of 10 reads from a minimum of 10 individuals). In total, the average percentage methylation across all sites was 35.5% $\pm$ 34.0 with a wide distribution (Figure \@ref(fig:sum-plot)A). After assigning CpGs hierarchically to promoter (within 2kb upstream of a TSS) > exon > intron, we found that 12.1% of sites were in promoters, 7.9% in exons, 11.8% in introns, and 68.1% in intergenic regions. At the level of genomic features, promoters had the lowest methylation (median = 5.3%, mean $\pm$ SEM = 20.5% $\pm$ 0.5), introns had intermediate methylation (median = 43.5%, mean $\pm$ SEM = 41.0% $\pm$ 0.5), and exons had the highest methylation (median = 54.3%, mean $\pm$ SEM = 46.7% $\pm$ 0.7). However, each of these features had a wide distribution of methylation percentages across different genes (Figure \@ref(fig:sum-plot)B).

*RRBS Data Analysis*

Output data from the sequence processing described above was analyzed in R version 4.1.1 [@rcite]. We processed the aligned sequence data with `MethylKit` version 1.14.2 [@akalin2012]. Using `MethylKit`, we extracted the number of total aligned reads and number of methylated or unmethylated reads for each CpG site. As recommended by the package authors, we filtered out CpGs above the 99.5th percentile to account for potential PCR bias.

For analyses of corticosterone and treatment associations, we filtered these CpGs to include only those that met the following criteria. First, we required a minimum coverage of 10 reads per sample to retain data for that sample at a given CpG. We further filtered the dataset to remove any CpGs that were mostly invariant (i.e., more than half of samples had methylation percentage of 0 or 100%) as well as CpGs that had extremely low variation [SD less than 5% across all samples, @husby2022; @lundregan2022]. For models comparing treatment effects, we required that females have data at a given CpG from both pre- and post-treatment sampling points to be included. We also required a minimum of six individuals per treatment group for a CpG to be included in the analysis. For basic descriptions of methylation patterns, we used all CpGs that had 10 reads or more in the pre-treatment samples.

The built-in differential methylation techniques in `MethylKit` are designed for two group comparisons with limited flexibility in modeling options. Because we had repeated measures before and after treatments for both groups, we could not specify the necessary models within `MethylKit` itself. Therefore, we exported and combined the filtered CpG records for all groups so that we could fit generalized linear mixed models (GLMMs) for each CpG site [as in @lindner2021] using the `glmer` function in R package `lme4` version 1.1-29 [@bates2014]. We fit a separate set of models for natural corticosterone variation before treatments occurred (baseline, stress-induced, or post-dexamethasone), for the within-year treatment effect, and for the between-year treatment effect. Each of these datasets were constructed separately since they included different subsets of both individual birds and of CpGs that met the criteria described above. 

For natural variation in corticosterone, we included only the pre-treatment samples. Using these samples, we fit a GLMM for each CpG with the number of methylated and unmethylated reads as the binomial response variable. We fit this set of models separately with baseline (n = 78,143 CpGs), stress-induced (n = 78,027 CpGs), or post-dexamethasone (n = 69,189 CpGs) corticosterone measured in ng/ml as the single continuous predictor variable. The models included a random effect for female identity to account for repeated sequencing of the same CpG sites within each female. We excluded the results for any models that failed to converge because we could not reliably estimate effects in those cases.

For within-year (n = 48,070 CpGs) and between-year (n = 6,787 CpGs) comparisons after treatments, we fit a single GLMM for each included CpG with the number of methylated and unmethylated reads as the binomial response variable. Predictors included pre-treatment methylation percentage at the CpG being modeled, a fixed effect of treatment (control vs. corticosterone), and a random effect for female identity. In each model, significance of the comparison between control and corticosterone treated birds was assessed using the `emmeans` version 1.5.3 package in R [@emmeans], which employs the Satterthwaite approximation method for binomial mixed models. We also evaluated the stability of methylation within individuals in these models by summarizing the regression coefficient of pre-treatment methylation on post-treatment methylation.

We accounted for multiple comparisons in each of these analyses by adjusting all p-values using the q-value approach implemented by the `qvalue` package version 2.20.0 in R with the false discovery rate (FDR) set at 0.05 [@qvalue]. The q-value approach accounts for multiple comparisons by considering both the number of comparisons made and the distribution of accumulated p-values. We only report and interpret estimates with q-values < 0.05. 

*Annotation of differentially methylated CpGs*

After identifying CpGs that were significantly associated with either natural corticosterone or experimental treatment with corticosterone, we identified genes associated with each CpG. We used the `bedtoolsr` package version 2.30.0-5 to select genes that had a significant CpG either within the gene body or within 2 kb upstream of the transcription start site [@bedtoolsr]. We generated separate lists of genes associated with CpGs for baseline corticosterone, stress-induced corticosterone, post-dexamethasone corticosterone, within-year treatment effects, and between-year treatment effects. For each of these comparisons we also generated a complete list of genes associated with all of the CpGs that passed the filtering criteria described above to be used as a null background list (see below).

Starting with the list of genes associated with each comparison set, we used the DAVID functional annotation tool [@DAVID] to test whether our genes were enriched in any molecular functions or biological processes in the Gene Ontology knowledgebase [@ashburner2000; @goterms]. For each comparison we used the custom background list generated above. This background list is important for interpretation because we were only able to test CpGs near a subset of genes in each comparison (number of genes included in testing for baseline corticosterone = 4,143; stress-induced corticosterone = 4,146; post-dexamethasone corticosterone = 3,863; within-year treatment = 2,913; between-year treatment = 452). 

Using DAVID we identified a set of GO terms associated with biological processes or molecular functions that were over represented in the list of significant CpGs compared to the background list for that comparison [@goterms]. We filtered this list to include only GO terms with p-values < 0.05 after applying a false discovery rate correction. We initially visualized the GO terms for each comparison using REVIGO [@revigo]; however, our study identified a relatively small number of GO terms and no clearly identifiable clusters of terms were identified in REVIGO. Therefore, we report the complete list of genes and GO terms associated with CpGs in each comparison.

## RESULTS

*Corticosterone and breeding timing*

Birds that were treated with corticosterone in year one had higher baseline corticosterone in year two (corticosterone treatment $\beta$ = 3.34; 95% confidence interval = 1.43 to 5.24; Figure \@ref(fig:carry-fig)A; Table \@ref(tab:carry-tab)). Corticosterone treatment in year one was not related to stress-induced corticosterone in year two (Table \@ref(tab:carry-tab)), but females that were exposed to corticosterone in year one had lower post-dexamethasone corticosterone in year two, indicating more robust negative feedback ($\beta$ = -4.86; 95% confidence interval = -9.01 to -0.70; Figure \@ref(fig:carry-fig)B; Table \@ref(tab:carry-tab)). Finally, corticosterone-exposed females initiated their nesting attempt earlier in the following year ($\beta$ = -2.34; 95% confidence interval = -4.49 to -0.20; Figure \@ref(fig:carry-fig)C; Table \@ref(tab:carry-tab)).

```{r carry-fig, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 3.5, fig.width = 7.3, fig.cap = "Effect of corticosterone treatment in one year on measures of baseline corticosterone (A), post-dexamethasone corticosterone (B), and clutch initiation date (C) in the following year. Points are partial residuals of raw data collected for average year effects. Boxes and whiskers show the median, interquartile range, and 1.5 times IQR."}

carry_fig <- readRDS(here::here("5_temporary_files", "carry_fig.rds"))
carry_fig

```

*Association between natural or experimental corticosterone and methylation*

Using pre-treatment samples, we found that methylation percentage at 116 CpGs out of 78,143 tested was significantly associated with baseline corticosterone after FDR correction (Figure \@ref(fig:volc-plot)A; Table  \@ref(tab:model-tab)). For stress-induced corticosterone, we found a similar significant association at 356 out of 78,027 CpGs that were tested (Figure \@ref(fig:volc-plot)B; Table \@ref(tab:model-tab)). For post-dexamethasone injection samples, we found a significant association between corticosterone and methylation at 735 out of 69,189 CpGs tested (Figure \@ref(fig:volc-plot)C; Table \@ref(tab:model-tab)).

In models examining the effect of corticosterone treatment, we found that for samples collected within the same breeding season 1-2 weeks after treatment, 111 out of 48,070 CpGs tested showed significant evidence of differential methylation between treatment and control groups after FDR correction (Figure \@ref(fig:volc-plot)D; Table \@ref(tab:model-tab)). We had fewer individuals and fewer CpGs that passed filtering for comparisons one year after treatment, but we found that 49 out of 6,787 CpGs tested were significantly differentially methylated between treatment and control groups after one year (Figure \@ref(fig:volc-plot)E; Table \@ref(tab:model-tab)). Although we were primarily interested in treatment effects, these models also showed that pre-treatment methylation at a given CpG site generally predicted post-treatment methylation both within a year (Figure \@ref(fig:rpt-sample)A) and for samples collected one year later (Figure \@ref(fig:rpt-sample)B).

```{r volc-plot, warning = FALSE, message = FALSE, fig.width = 8, fig.height = 6, fig.cap = "Association between DNA methylation and corticosterone from GLMMs based on observational and experimental study components. Panels A, B, and C show the pre-treatment regression coefficient for baseline corticosterone, stress-induced corticosterone, and post-dexamethasone corticosterone on methylation percentage, respectively. Panels D and E show the difference in methylation for control vs. treatment groups after accounting for pre-treatment methylation percentage for samples 1-2 weeks after treatment (D) and 1 year after treatment (E). In all plots, -log base 10 p-values are shown on the y axis with red points indicating CpGs that were significantly associated with corticosterone after applying the false discovery rate correction. Horizontal dashed lines indicate p-values of 0.05, 0.01, 0.001, and 0.0001 moving from the bottom to top of each plot to aid in interpretation."}
pviol <- readRDS(here::here("5_temporary_files/pviol.rds"))

plot(pviol)

```

*Association between differentially methylated CpGs and genes*

We found that CpGs that were significantly associated with baseline corticosterone, stress-induced corticosterone, and post-dexamethasone corticosterone were located in or near a total of of 32, 176, and 236 identifiable genes, respectively (Table \@ref(tab:gene-table)). When comparing differentially methylated CpGs after treatment effects, within-year and between-year CpGs were located in or near 52 and 16 genes, respectively (Table \@ref(tab:gene-table)). A subset of these genes were identified in two or three different comparisons (Figure \@ref(fig:gene-venn)). Because of our filtering process many genes were not tested in each comparison (i.e., the background set of possible genes tested differed for each comparison).

```{r gene-venn, echo = FALSE, message = FALSE, warning = FALSE, out.width = "95%", fig.align = "left", fig.cap = "Number of genes near CpGs that were significantly associated with natural variation in corticosterone (baseline, stress-induced, and post-dexamethasone) or with experimental corticosterone elevation (within-year and between-year). No identified genes were shared between the comparisons with circles that do not overlap."}
knitr::include_graphics("genes.png")

```

In examining the function of genes identified in this process, we found no clear evidence for direct links to genes known to be involved in regulation of the HPA axis. Stress-induced corticosterone was linked to a single CpG associated with MC2R, which encodes the ACTH receptor, but closer examination of the CpG in question suggested that this association resulted from similar proteins included in our genome annotation and more likely gene was MC3R, which is not directly involved in the HPA axis. Several genes known to be associated with the HPA axis (e.g., CRH, CRHR1, FKBP5) did not have any CpGs near them in the background set, so we could not test for differences associated with these genes.

*GO term analysis*

Using the gene lists from Table \@ref(tab:gene-table) as input, we identified GO terms that were significantly associated with each comparison. With the false discovery rate set at 0.05, we identified 14 GO terms associated with baseline corticosterone, 22 terms associated with stress-induced corticosterone, 10 terms associated with post-dexamethasone corticosterone, 2 terms for the within-year treatment effect, and 27 terms for the between-year treatment effect (Table \@ref(tab:go-table)). None of these lists resulted in any clear clustering of processes using the REVIGO visualization tool and many terms were repetitive and attributable to the same few gene associations. 

Baseline corticosterone was associated with photoreceptor activity and response to light, which was primarily driven by opsin and rhodopsin gene associations (OPN1SW, RHO, LWS). Stress-induced corticosterone was associated with a wider range of processes connected to a larger set of genes. These included a variety of cell signaling and receptor pathways (e.g., MC2R, MC3R). Post-dexamethasone corticosterone was primarily associated with signaling receptor activity driven by a relatively large number of associated genes (Table \@ref(tab:go-table)).

Differentially methylated CpGs for within-year corticosterone treatment were only related to two GO terms associated with structural cell components and attributable to genes of unknown function. Between-year corticosterone treatment was associated with a variety of GO terms having to do primarily with transmembrane receptor signaling, but nearly all of these terms were selected from the same set of gene associations (BMPR1A and B, ACVR1, and TGFBR1). 

## DISCUSSION

We found that experimental increases in corticosterone induced long-term phenotypic changes. Females that experienced a few brief spikes in exogenous corticosterone had stronger negative feedback in the HPA axis and bred earlier in the subsequent year; these characteristics are typically associated with high stress resilience and reproductive success in this population. Furthermore, natural variation in corticosterone was correlated with DNA methylation, and experimental treatments altered DNA methylation patterns. Importantly, regulation of DNA methylation in response to corticosterone occurred rapidly in adults (within days) and resulted in detectable changes at least one year after treatment, paralleling the changes in physiological and behavioral phenotypes. Taken together, these results support the idea that the activation of the stress response machinery changes traits associated with stress resilience, and thus may prime future responses to challenges. DNA methylation could act as a key mechanism linking the prior experience of stressors–including during adulthood–to subsequent coping ability. Rapid endocrine flexibility and adaptive calibration of the stress response have emerged as key determinants of resilience to challenges [@taff2016; @grindstaff2022; @hau2015] and understanding the mechanistic basis of these patterns is an important step in predicting when flexibility is sufficient for coping with changing conditions.

The changes in phenotype that we detected one year after experimentally elevating corticosterone partially matched our predictions if exposure altered phenotype in ways that would increase future stress resilience. We found that, compared to controls, experimental females initiated breeding earlier and had stronger negative feedback in the subsequent year. In tree swallows, clutch initiation date is a strong predictor of both seasonal and lifetime reproductive success and is often considered a proxy for individual quality or condition [@winkler2020]. Similarly, the strength of negative feedback is consistently the best physiological predictor of coping ability and reproductive success both under natural conditions and after imposing experimental challenges [@taff2018; @zimmer2019]. However, contrary to our prediction, treatment had no effect on stress-induced corticosterone the following year. We also found that compared to control females, corticosterone dosed females had higher baseline corticosterone and no difference in stress-induced corticosterone one year after treatment. Baseline corticosterone does not predict stress resilience in this population [@zimmer2019]. However, because baseline corticosterone often increases in preparation for periods of high energetic demands, including the demands of reproduction [the cort-adaptation hypothesis, @bonier2011; @hau2010; @casagrande2018], these results might reflect an increased allocation to breeding in subsequent years in corticosterone treated females. For example, female European starlings (*Sturnus vulgaris*) manipulated to increase parental investment increased their baseline corticosterone during incubation [@love2014]. Similarly, tree swallows that increase baseline corticosterone more over the reproductive period provision offspring at higher rates [@bonier2011]. Thus, our results might represent a combination of priming effects coupled with the immediate energetic demands of breeding earlier. Alternatively, corticosterone treatment in year one could have influenced measurements in year two indirectly, rather than by priming responses to later challenges.

Our study also adds to the growing recognition of bidirectional links between coping ability and DNA methylation. While this relationship has been demonstrated in laboratory-based model systems [@liu1997; @weaver2004] the potential for environmental stressors to trigger methylation, and affect subsequent coping ability, has only recently been explored in wild animals. Early results in wild animals suggest patterns similar to those seen in laboratory rodents. For example, early life maternal care and social connections in spotted hyenas (*Crocuta crocuta*) predict DNA methylation and glucocorticoid regulation as an adult [@laubach2019; @laubach2021]. Similar effects can play out in adulthood; for example, in savannah baboons (*Papio cynocephalus*) high social status as an adult is associated with more rapid changes in DNA methylation (epigenetic aging) as a consequence of the social stress that accompanies high status [@anderson2021]. Our results are consistent with the results derived from lab rodents, wild mammals, and a few studies in wild birds [e.g., @rubenstein2016], suggesting that flexible adjustment of methylation may be a general mechanism by which prior experiences of stressors are encoded in order to modulate future responses to challenges. 

While there has been a rapid increase in studies of methylation in wild birds in recent years [e.g., @vonholdt2022; @rubenstein2016; @makinen2019, @sheldon2018], relatively few studies have sampled the same adults multiple times. Our study design allowed us to assess the stability of genome-wide DNA methylation within individuals. We found that many CpGs that we interrogated had large between individual differences in methylation and that those differences were typically stable even in samples collected one year apart. Compared to these individual differences, flexible changes in methylation were relatively smaller and detectable at fewer CpGs. The stable individual differences that we detected may be largely explained by genetic differences between individuals [e.g., @sepers2023], but the consequences of early life conditions may also play an important role in generating differences [@laubach2019; @jimeno2019; @sheldon2018]. For example, early life climate conditions are related to lifelong methylation of the glucocorticoid receptor gene in superb starlings [*Lamprotornis superbus*, @rubenstein2016]. We could not assess the possibility of a similar pattern in our study because we did not have any information on early life conditions for our birds. Regardless of the source of these initial differences, our results clearly demonstrate that detecting subtle adjustments of methylation in adulthood to any treatment of interest will often require accounting for pre-treatment methylation. Our results also suggest that if changes in methylation are ecologically relevant for phenotypic flexibility during adulthood, these effects are likely to occur through modest changes at specific locations rather than through large scale modifications to methylation patterns.

At this point, any functional consequences of the specific methylation changes that we detected are unclear. None of the genes or GO terms associated with natural variation in corticosterone or with treatments had clear connections to HPA axis regulation. We previously found that non-specific, genome-wide methylation predicts stress resilience to experimental challenges in this population [@taff2019]. Thus, the differences that we detected might reflect large-scale regulation of methylation rather than targeted regulation of sites with specific functional consequences. Alternatively, some of the changes that we detected might have functional effects on stress response calibration that are not obvious from the known effects of those genes. In support of this idea, we did find some overlap between methylation in the genes associated with natural variation in corticosterone and with the  consequences of our experimental manipulation of corticosterone. In particular, post-dexamethasone corticosterone, which is a strong predictor of stress resilience in tree swallows [@taff2018; @zimmer2019], had the most extensive correlations between methylation and identified genes and some of these genes were shared with the other corticosterone measures and with treatment effects. 

Another potential reason for our failure to find clear links between changes in DNA methylation and genes associated with the stress response may result from limitations of our approach. An advantage of RRBS is that it does not rely on pre-selecting candidate genes, but a disadvantage is that not all relevant genes are necessarily tested. After filtering our data, many of the genes with known roles in the HPA axis were not included in comparisons or had coverage at only a few CpG sites. Thus we did not directly test for methylation differences for many key genes. It is possible that deeper sequencing of our libraries would have improved our ability to detect functional differences. Moreover, although we used the most complete reference genome available for tree swallows, many CpGs mapped to predicted genes whos function is unknown. Continued improvement of assembly and annotation for reference genomes of non-model organisms is important for understanding the functional importance of epigenetic changes. Studying DNA methylation in non-model systems is a rapidly developing field and many recent papers outline the pros and cons of various approaches [@sepers2019; @laine2022; @beck2022]. One particularly promising approach that may strike a balance between a focus on candidate genes and the ability to detect genome-wide associations is to combine RRBS with probes that enrich sequences at a potentially large number of target genes [target-enriched enzymatic methyl sequencing, @rubenstein2023].

Regardless of the functional consequences of the changes we detected, we found that brief increases in corticosterone have effects on subsequent corticosterone regulation, breeding decisions, and methylation a full year after dosing ended. At least some of the phenotypic changes we detected support a hormone-mediated priming effect in which activation of the stress response machinery improves the capacity to cope with future challenges, increasing stress resilience. The fact that these changes in phenotype are coupled with changes in methylation patterns implicates the regulation of DNA methylation as a potential mechanism of flexibly adjusting the stress response system based on prior experiences, although our study cannot determine whether changes in methylation directly cause adaptive changes in the subsequent stress response and will need to be paired with analyses of gene expression. Understanding the mechanisms that integrate experience with future stress responsiveness has important consequences for predicting how and when individuals can cope with repeated exposure to challenges. Conceptual models of the stress response suggest that while repeated challenges can sometimes impose costs [@romero2009; @mcewen2003], calibration achieved through regular activation of the stress response may prime more effective responses to future challenges [@del2011; @hilker2016]. Studying the mechanisms by which stress exposure is encoded biologically will help to differentiate these possibilities and shed light on when and how individuals succeed or fail through flexible regulation of the physiological response to challenges.

*Data and code availability*

The complete set of bioinformatic processing scripts, R code, and data associated with each sample is available and permanently archived on Zenodo (https://doi.org/10.5281/zenodo.8125153). Raw sequence data from RRBS is available on GenBank (BioProject ID PRJNA953597).

## ACKNOWLEDGMENTS

We thank the members of the tree swallow research team for assistance with fieldwork including, Joseph Byington, Joe Colcombe, Collin Dickerson, Garret Levesque, Jacob Kolenda, Lyra Liu, Sophie Nicolich-Henkin, Teresa Pegan, Avram Pinals, Alyssa Rodriguez, Vanesa Rodriguez-Arcila, Tom Ryan, David Scheck, Lauren Smith, Jocelyn Stedman, and Cedric Zimmer. Bronwyn Butcher and the Lovette Lab provided input on lab methods and members of the Vitousek Lab provided feedback on earlier versions of the manuscript. Charlotte Holden produced the tree swallow illustrations. 

## AUTHOR CONTRIBUTIONS

CCT and MNV conducted the field based data collection. CCT and MNV conceived the study. CCT and SMM conducted the lab work for RRBS. CCT and LC conducted the lab work for creating the reference genome and LC carried out the bioinformatics for genome assembly and annotation. CCT analyzed and visualized the data with assistance from SMM and LC. CCT drafted the paper with input from all authors.

## ETHICAL NOTE

All methods were approved by Cornell IACUC and sampling was conducted with appropriate state and federal permits.

## FUNDING

Research was supported by DARPA D17AP00033 and NSF-IOS grants 1457251 and 2128337. The views, opinions and/or findings expressed are those of the authors and should not be interpreted as representing the official views or policies of the Department of Defense or the U.S. Government. CCT was supported by DARPA D17AP00033 and by the Rose Postdoctoral Program at the Cornell Lab of Ornithology and SMM was supported by the Rose Postdoctoral program at the Cornell Lab of Ornithology.

## REFERENCES

<div id="refs"></div>

\newpage
\beginsupplement

## SUPPLEMENTARY FIGURES AND TABLES


```{r read-summary, message = FALSE, warning = FALSE, out.width = "95%", fig.align = "left", fig.cap = "Summary of sequencing and methylation call results from raw sequence data. Panel A shows the distribution of the total number of sequences for each sample and number of sequences that aligned to the tree swallow genome. Panel B shows the number of CpH sites that were methylated or unmethylated for each sample. Panel C shows the number of CHH sites that were methylated or unmethylated for each sample. Panel D shows the number of CpG sites that were methylated or unmethylated for each sample. Panel E shows the percentage of total CpG reads that were methylated by sample. Note that these histograms are based on raw sequencing results that do not account for differential coverage between samples or locations in the genome and are included for illustration only."}

knitr::include_graphics("read_summary.png")

```

```{r sum-plot, message = FALSE, warning = FALSE, fig.width = 7.6, fig.height = 4.2, fig.cap = "Distribution of methylation percentage across all CpGs and for different genomic features. Panel A shows entire distribution of methylation percentage for all 148,167 CpGs from pre-treatment samples before any filtering. Panel B shows the methylation percentage for exons, introns, and promoters that had CpGs identified within them. Horizontal lines, boxes, and whiskers show the median, interquartile range, and 1.5 times IQR respectively. The black circle within each box is the mean for that feature."}

sum_plot <- readRDS(here::here("5_temporary_files/sum_plot.rds"))

sum_plot

```

\newpage

```{r rpt-sample, message = FALSE, warning = FALSE, fig.width = 7.4, fig.height = 3.8, fig.cap = "Association between pre-treatment methylation percentage and post-treatment methylation at each CpG for samples collected within a breeding season (panel A) and for samples collected one year apart (panel B). To help with interpretation, horizontal dashed lines indicate p-values of 0.05, 0.01, 0.001, and 0.0001 moving from the bottom to top of the plots."}

rpt_sam <- readRDS(here::here("5_temporary_files/consistent_meth.rds"))
rpt_sam

```


\newpage 
\pagebreak
 
 

```{r model-tab, warning = FALSE, message = FALSE}

mod_tab <- data.frame(comparison = c("Baseline corticosterone", "Stress-induced corticosterone", "Post-dexamethasone corticosterone",
                                     "Within year treatment", "Between year treatment"),
                      CpGs = c(78143, 78027, 69189, 48070, 6787),
                      n = c("29.2 +/- 7.0", "30.6 +/- 7.3", "14.2 +/- 3.0", "24.5 +/- 8.8", "9.8 +/- 1.5"),
                      #sd = c(7.0, 7.3, 8.8, 1.5),
                      sig = c(116, 356, 735, 111, 49),
                      genes = c(32, 176, 236, 52, 16),
                      go = c(14, 22, 10, 2, 27))

library(magrittr)
knitr::kable(mod_tab, col.names = c("Comparison", "CpGs evaluated", "Individuals per comparison",
                                    "Significant CpGs", "Associated Genes", "Significant GO terms"),
             caption = "Summary of GLMMs for each comparison with the number of CpGs significantly correlated with corticosterone or differentially methylated between treatment groups. One model was fit for each CpG site; see text for description of models.",
             align = c("l", "r", "r", "r", "r", "r")) %>%
  kableExtra::kable_styling(full_width = TRUE) %>%
  kableExtra::kable_styling(latex_options = c("striped", "hold_position"))
  

```

\newpage 
\pagebreak
 
 

```{r carry-tab, message = FALSE, warning = FALSE, echo = FALSE}
tabrs <- read.delim(here::here("5_temporary_files/carry_tab.txt"), header = FALSE)

kableExtra::kbl(tabrs, caption = "Results of GLMs for corticosterone and clutch initiation date in the year after experimental treatments were applied.", 
    col.names = c("Predictor", "Estimate", "CI", "P"), booktabs = T) %>%
  kableExtra::kable_styling(latex_options = c("striped", "repeat_header", "hold_position")) %>%
  kableExtra::pack_rows("Baseline corticosterone (n = 37)", 1, 4) %>%
  kableExtra::pack_rows("Stress-induced corticosterone (n = 37)", 5, 9) %>%
  kableExtra::pack_rows("Post-dexamethasone corticosterone (n = 33)", 10, 14) %>%
  kableExtra::pack_rows("Clutch initiation date (n = 41)", 15, 18)
```

\newpage 
\pagebreak
 
 

```{r gene-table, echo = FALSE, message = FALSE, warning = FALSE}
gene_table <- data.frame(
  Comparison = c("Baseline corticosterone", "Stress-induced corticosterone", "Post-dexamethasone corticosterone", "Treatment within year", "Treatment between years"),
  Genes = c(
    
    "ABHD8; CBX2; CCDC15; CLP1; HELZ; HMGA1; LIG3; LOC100220428; LOC100221958; LOC100223410; LOC100228079; LOC751972; LWS; MAST3; NFASC; NRCAM; OPN1SW; PCBP2; PCBP3; PCBP4; PCMT1; PUS3; RHO; SESN1; SESN3; STAMBPL1; STMN1; STMN2; STMN3; TAF6L; TMIE; ZFYVE19",
            
            "ACVR1; ACVR1B; ACVRL1; ADGRB1; ADGRB2; ADGRB3; AKT1; AKT3; ALPK1; ALX1; ARHGEF11; ARID3A; ARID3B; ARID3C; ASB3; AXIN1; AXIN2; BMPR1A; BMPR1B; CLDN1; CLDN14; CLDN5; COPA; DCLK1; DCX; ESRRG; FOSL2; FOXJ1; GALM; GATA4; GATA5; GATA6; GPR171; HID1; HNF4A; HNF4G; HOXA3; HOXB3; HSPA13; HSPA14; HSPA2; HSPA5; HSPA8; HSPA9; IER5L; IGHMBP2; IGSF9B; IKZF1; IKZF2; IKZF3; ITGA11; KIF1A; KIF1B; KLHL12; KLHL17; KLHL18; KLHL2; KLHL20; KLHL5; LAMB1; LAMB2; LAMB3; LDLRAP1; LECT2; LOC100217876; LOC100219031; LOC100220115; LOC100221543; LOC100222291; LOC100222941; LOC100223643; LOC100224585; LOC100224644; LOC100224843; LOC100225871; LOC100226582; LOC100226815; LOC100227201; LOC100227468; LOC100227703; LOC100229020; LOC100229354; LOC100229528; LOC100229609; LOC100229630; LOC100230278; LOC100230328; LOC100230362; LOC100231693; LOC100231893; LOC100232259; LOC101233849; LOC105758838; LOC105759110; LOC105759301; LOC105759399; LOC105759510; LOC105759937; LOC115491093; LOC115491353; LOC115491354; LOC115491358; LOC115491417; LOC115491512; LOC115494566; LOC115494567; LOC115496063; LOC115496268; LOC115497065; LOC115497066; LOC115497067; LOC115497068; LOC115497095; LOC115497281; LOC115498367; LOC116807418; LOC116807419; LOC121468018; MAPK1IP1L; MC1R; MC2R; MC3R; MC4R; MC5R; MEX3A; MEX3B; MEX3D; MITF; MNX1; MOB1B; MOB3C; MYO1D; NCOA5; NCOR2; NKX3-2; NR2E1; NR2E3; NR2F1; NR2F2; NR2F6; NR5A2; P2RY12; P2RY13; P2RY14; PCDH10; PCDH12; PCDH8; PIM1; PIM3; PKNOX1; PKNOX2; PLEKHA6; PLEKHO1; PTCH1; PTK7; RAX; RAX2; RXRA; SGK1; SGK2; SIX6; SLC25A10; SLC66A1; SRSF3; SYNPO; TFEB; TFEC; TGFBR1; TIMM8A; TNRC18; UBALD1; UBALD2; UBXN7; WDR7; ZCCHC14; ZDHHC23",
    
    "ACVR1; ACVR1B; ACVRL1; ADORA1; ADORA2A; ADORA2B; ADRA1B; AK8; ATOH7; BMPR1A; BMPR1B; C18H17orf58; CBX2; CCM2; CCNA1; CHGB; CHL1; CHST11; CHST12; CHST13; CLIP1; COPA; CORO2A; CORO2B; CRTC2; CRYBB1; CTRL; CTXN1; CTXN2; DENND4C; DNAJB13; DNMT3B; DOCK7; DOCK8; EMX2; ERICH3; ESRRG; EVA1B; FAM83A; FGF18; FGF8; FZD1; FZD10; FZD2; FZD3; FZD4; FZD5; FZD6; FZD7; FZD8; GAD1; GAD2; GATA2; GATA3; GNPTG; GPR171; GPR26; GPR78; GPR83; GRB2; HID1; HNF4A; HNF4G; ID=cds-NP_001232480.1; ID=cds-NP_001232635.1; ID=cds-NP_001243137.1; ID=cds-NP_001243138.1; ID=cds-XP_030112754.3; ID=cds-XP_030134736.2; ID=cds-XP_032603585.2; IGHMBP2; IKZF1; IKZF2; IKZF3; ILF3; IP6K1; IP6K2; IP6K3; ITGA11; KATNAL2; KCNA10; KCNA2; KCNA4; KCNA5; KDM5A; KHDRBS1; LOC100190025; LOC100218025; LOC100219450; LOC100220115; LOC100221543; LOC100221838; LOC100221958; LOC100223460; LOC100224073; LOC100225784; LOC100225871; LOC100226496; LOC100226582; LOC100226723; LOC100227201; LOC100227464; LOC100227703; LOC100229528; LOC100229609; LOC100229630; LOC100229947; LOC100230362; LOC100231785; LOC100232259; LOC101232904; LOC101232979; LOC101233511; LOC105758604; LOC105758698; LOC105758845; LOC105758859; LOC105759399; LOC105759510; LOC115490659; LOC115492615; LOC115494788; LOC115496018; LOC115496268; LOC115496876; LOC115497044; LOC115497045; LOC115497065; LOC115497066; LOC115497067; LOC115497068; LOC115497095; LOC115498277; LOC115498450; LOC115498504; LOC115498524; LOC116806919; LOC116807124; LOC116807173; LOC116807372; LOC116807374; LOC116807475; LOC116807566; LOC116807567; LOC116807569; LOC116807570; LOC116807571; LOC116807574; LOC116807590; LOC116809047; LOC116809297; MEIS1; MEIS2; MEX3A; MEX3B; MEX3D; MGAT5; MNX1; MOV10; MYH10; MYH11; MYH9; MYO5A; NANOS3; NEUROG1; NEUROG2; NFASC; NKX6-1; NKX6-2; NR2E1; NR2E3; NR2F1; NR2F2; NR2F6; NR5A2; NRCAM; NUAK1; NUAK2; P2RY12; P2RY13; P2RY14; PALB2; PAN2; PAPSS1; PAPSS2; PATL1; PCDH10; PCDH12; PCDH18; PCDH8; PCMT1; PFKL; PI15; PIK3C2B; PIM1; PIM3; PLEKHA6; PTCH1; QRFPR; R3HDML; RPL13A; RXRA; S100A11; SAMD11; SEC16A; SEPTIN11; SEPTIN2; SEPTIN3; SEPTIN5; SEPTIN6; SEPTIN8; SH3GL2; SH3GL3; SLC2A6; SLC38A10; SMARCC1; SPOPL; STX8; SYNGR3; SYT7; TAL2; TCF7L2; TGFBR1; TRPM2; TWF1; TWF2; VIPR1; VPS26A; VPS26B; VWCE; WIPI1; WIPI2; WWP2; ZDHHC5; ZDHHC8; ZNF598",
            
    "ACTA1; ACTA2; ACTB; ACTC1; ACTG1; ACTG2; ACTR1A; ASS1; ESPN; GALNT3; GALNT6; GMPS; GPR27; GPR85; INHBB; IRF7; ITGA11; KCNG1; KCNG2; KCNG4; LOC100190135; LOC100218246; LOC100218887; LOC100220115; LOC100220137; LOC100227878; LOC105758845; LOC105758846; LOC105758847; LOC105758850; LOC105758859; LOC105760754; LOC115494114; LOC115496608; LOC115497065; LOC115497066; LOC115497067; LOC115497068; LOC115497975; LOC115498396; LOC115498513; LOC115498523; LOC116809297; NTN1; NTN3; NUAK1; NUAK2; PCDH15; RIMS3; TMPRSS6; VDAC1; VDAC2",
    
    "ACVR1B; ACVR1; ACVRL1; BMP1; BMPR1A; BMPR1B; LOC115495659; LOC115495665; LOC115495661; LOC115490544; MMADHC; NETO1; TBKBP1; TGFBR1; TLL2; TLL1")
)
library(magrittr)
library(knitr)
kable(gene_table, booktabs = TRUE, longtable = TRUE, caption = "List of genes with differentially methylated CpGs in exons, introns, or within 2kb upstream of the gene.") %>%
  kableExtra::kable_styling(full_width = TRUE) %>%
  kableExtra::column_spec(1, width = "3cm") %>%
  kableExtra::kable_styling(latex_options = c("repeat_header", "striped", "hold_position"))

#%>%
 # kableExtra::kable_styling(full_width = TRUE, longtable = TRUE) %>%
  #kableExtra::column_spec(2, width = "12cm")

```

\newpage 
\pagebreak
 
 

```{r go-table, echo = FALSE, message = FALSE, warning = FALSE}
go_table <- readRDS(here::here("4_other_output", "go_table.rds"))
go_tab2 <- go_table[, c("GO_term", "FDR2", "Function")]

kable(go_tab2, booktabs = TRUE, longtable = TRUE, caption = "GO terms identified using DAVID after correction to a false discovery rate of 0.05 for each comparison set.", col.names = c("GO Term", "FDR", "Function")) %>%
  kableExtra::kable_styling(latex_options = c("striped", "repeat_header", "hold_position")) %>%
  kableExtra::pack_rows("Baseline corticosterone", 1, 14) %>%
  kableExtra::pack_rows("Stress-induced corticosterone", 15, 36) %>%
  kableExtra::pack_rows("Post-dexamethasone corticosterone", 37, 46) %>%
  kableExtra::pack_rows("Within-year treatment", 47, 48) %>%
  kableExtra::pack_rows("Between-year treatment", 49, 75)

```

\newpage
\pagebreak

```{r trt-plot, message = FALSE, warning = FALSE, fig.width = 5.5, fig.height = 5.2, fig.cap = "Effect of treatment variations on (A) baseline corticosterone and (B) stress-induced corticosterone 1-2 weeks after treatments ended. Each application of corticosterone results in a single spike in circulating levels that lasts a few hours. Repeated treatment during incubation results in a change in endogenous regulation of corticosterone levels. Individuals differ in the magnitude of this shift, but all corticosterone exposed treatments are categorically different than control treatments. For this study, corticosterone exposed and control treatments were grouped to simplify comparisons. See text for description of each treatment variation."}

trt_plot <- readRDS(here::here("5_temporary_files/treat_eff_fig.rds"))

trt_plot

```

\newpage
\pagebreak

```{r n-tab, cho = FALSE, message = FALSE, warning = FALSE}

ntable <- data.frame(
  trt_cat = c("Control", "Control", rep("Corticosterone Exposed", 5), "None", "Control", "Control", rep("Corticosterone Exposed", 3)),
  smp_type = c("Sham", "Full Control", "High", "Long1", "Long2", "Low", "Short", "No Manipulation", "Sham", "Full Control", "High", "Long1", "Low"),
  wthn_n = c(rep(NA, 7), 45, 10, 10, 9, 6, 10),
  btwn_n = c(4, 11, 10, 6, 2, 8, 4, NA, 3, 5, 8, 4, 7)
  )

kable(ntable, booktabs = TRUE, longtable = TRUE, caption = "Sample sizes for the comparisons in main analyses. See text for description of each treatment variation.", col.names = c("Treatment Category", "Sample Type", "Within-Year n", "Between-Year n")) %>%
  kableExtra::kable_styling(latex_options = c("striped", "repeat_header", "hold_position")) %>%
  kableExtra::pack_rows("Between-year phenotypic effects of corticosterone dosing", 1, 7) %>%
  kableExtra::pack_rows("Correlation between methylation and natural corticosterone", 8, 8) %>%
  kableExtra::pack_rows("Effects of corticosterone dosing on methylation", 9, 13)

```

